### PR TITLE
[GSIP-199] Document process to add service providers

### PR DESCRIPTION
**Pull request by jodygarnett**
[GSIP-199](https://osgeo-org.atlassian.net/browse/GSIP-199) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geoserver&utm_campaign=badge_info)

Document the process to add a service provider to our website, and associated osgeo project page.

Please see https://github.com/geoserver/geoserver/wiki/GISP-199 for details.

## Checklist

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the `main` branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

Developer documentation process documentation not visible to end users.

### CODE REVIEW

File: doc/en/developer/source/policies/service_providers.rst [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+In addition:
+
+* Project `sponsorship <https://github.com/geoserver/geoserver/wiki/Sponsor>`__ is another way to increase visibility, with specific activities available to sponsor throughout the year. Financial support is also recognized on the `osgeo sponsor <https://www.osgeo.org/sponsors/>`__ page.
+
+Service Provider checklist
+--------------------------
+
+To recognizing experienced providers and core contributors consistently we have developed the following checklist.
+
+Core contributors:
```
Comment (aaime) [inline doc/en/developer/source/policies/service_providers.rst:24]: Core contributors are doing all of these

File: doc/en/developer/source/policies/service_providers.rst [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+* ``core`` demonstrating an ongoing responsibility: security email list, releases, supported extension
+* ``committer`` active development: geoserver-devel, review of pull-requests, assigned jira tickets, development activities such bug-stomps and sprint participation
+* ``support`` sustained community support: geoserver-user, review of jira, gis.stackexchange.com, something!
+
+Experienced providers: work on GeoServer when hired by customers but are not always active
```
Comment (aaime) [inline doc/en/developer/source/policies/service_providers.rst:30]: However experienced providers are doing only some of these, right?

File: doc/en/developer/source/policies/service_providers.rst (hunk 1 above)
Comment (jodygarnett) [reply doc/en/developer/source/policies/service_providers.rst:24]: These three all have an aspect of taking responsibility for the geoserver project so I am marking them core.

File: doc/en/developer/source/policies/service_providers.rst (hunk 2 above)
Comment (jodygarnett) [reply doc/en/developer/source/policies/service_providers.rst:30]: The goal here is to allow folks to demonstrate (preferability with a link) the kind of work they are good at.

Experienced providers may do some of these, and even help out with releases on occasion, the difference is they are not taking on ongoing responsibility.

**Pull request merged**